gossipsub.go

```diff
@@ -348,6 +352,17 @@ func (gs *GossipSubRouter) AddPeer(p peer.ID, proto protocol.ID) {
 	if direct {
 		gs.p.host.ConnManager().TagPeer(p, "pubsub:direct", 1000)
 	}
+
+	// track the connection direction
+	outbound := false
+	conns := gs.p.host.Network().ConnsToPeer(p)
+	for _, c := range conns {
```

Comment:
This is perhaps overly paranoid, but is it possible to only count the connection used for the gossipsub stream? Otherwise there's an outside chance someone could trick us into open some other libp2p conn to them and get counted as outbound. Not sure how, exactly, although if we're running the autonat service they could request a dial-back.

Reply:
I guess we could, but it sounds too paranoid?

Reply:
we can however.

Reply:
Also, autonat dialback is not an issue -- we don't use the same host for the dial back.

Reply:
done.